### ISSUE TITLE
Misleading ImportError is raised when multiple Parallel objects are created in a with statement

### ISSUE THREAD

**Issue by ShadowTemplate** [joblib/joblib#632]
Hi! An _ImportError_ is raised whenever I try to instantiate two _Parallel_ objects in a _with_ statement.
Here is a minimal example to reproduce the situation:

```python
from joblib import Parallel


def main():
    with Parallel(-1), Parallel(-1):
        pass


if __name__ == "__main__":
    main()
```

joblib version: 0.11; OS: Linux; Python interpreter: 3.6.4

I believe the exception is a little bit misleading, since my system supports forking:
```
ImportError: [joblib] Attempting to do parallel computing without protecting your import on a system that does not support forking. To use parallel-computing in a script, you must protect your main loop using "if __name__ == '__main__'". Please see the joblib documentation on Parallel for more information
```
Is this a bug or the library is designed to support at most once _Parallel_ object with those properties?

The exception is raised [here](https://github.com/joblib/joblib/blob/master/joblib/_parallel_backends.py#L392) the second time the _configure_ method of _MultiprocessingBackend_ class is called (_already_forked_ becomes True).

Thanks for your attention and your awesome work.

**Comment by lesteve** [comment]
Very likely a duplicate of https://github.com/joblib/joblib/issues/334. I am going to close this issue, feel free to shout if you strongly disagree.

Out of interest can you give some details about what your use case is and why you need to create two different `Parallel` objects?

**Comment by ShadowTemplate** [comment]
I also get the same error of the linked issue since my processor has only 4 cores and the example uses 3+2 _num_jobs_.

My use case: I have two kinds of tasks T1 and T2. I need to run them alternatively via _Parallel_ many times (T1, T2, T1, T2, ...).T1 requires much more time than T2, so their optimal _batch_size_s are different.  If I use only one _Parallel_ object it will fail to consistently use the optimal size for each task once learnt: it will auto-tune for T1, be inefficient for T2 and re-auto-tune for T2, be inefficient for T1 and re-auto-tune and so on...

**Comment by lesteve** [comment]
Thanks for the details! The auto-tuning problem that you are seeing may have been fixed by https://github.com/joblib/joblib/pull/556 that resets the auto-tuned batch_size on each Parallel call. It would be great if you could try using joblib master and let us know!

**Comment by lesteve** [comment]
Thanks for following up on this @ShadowTemplate! I am reasonably confident that this was fixed in #556 and it looks like you agree.
